layer/layer_store.go

```diff
@@ -36,6 +37,8 @@ type layerStore struct {
 
 	mounts map[string]*mountedLayer
 	mountL sync.Mutex
+
+	locker *locker.Locker
```

Comment:
it's common to put locks above the fields they are protecting (perhaps a comment would help as well, describing what the locks are for)

Reply:
I did had a comment in some version :) but it got lost in the turmoil. Fixed.